api/renter.go

```diff
@@ -245,6 +245,23 @@ func (api *API) renterContractsHandler(w http.ResponseWriter, _ *http.Request, _
 	})
 }
 
+func (api *API) renterContractDetailHandler(w http.ResponseWriter, req *http.Request, ps httprouter.Params) {
```

Comment:
convention is to match the handler name to the route. So `/renter/contracts/:id` would become `renterContractsIdHandler`, where a capital letter indicates a `/`. Since we haven't fully settled on the route name yet, find to leave it.

This also needs a docstring, even if the docstring is pretty simple.